### PR TITLE
assert that ES is empty when deleting a domain

### PR DESCRIPTION
**Pull request by nickpell**
Adding this check because some previously deleted case data was still in ES for enikshay, and was not removed by deleting other cases.

We would want to know that data still is in ES, if it requires manual intervention to clear.

### CODE REVIEW

**Comment by nickpell** [comment]
✅

File: corehq/apps/domain/management/commands/delete_domain.py [hunk 1]
```diff
@@ -40,6 +41,11 @@ def handle(self, domain_name, **options):
             if confirm != domain_name:
                 print("\n\t\tDomain deletion cancelled.")
                 return
+
         print("Deleting domain {}".format(domain_name))
         domain.delete()
+
+        for hqESQuery in [AppES, CaseES, CaseSearchES, DomainES, FormES, GroupES, LedgerES, UserES]:
```
Comment (orangejenny) [inline corehq/apps/domain/management/commands/delete_domain.py]: Is there, or should there be, a way to get the list of all ESs?

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/domain/management/commands/delete_domain.py [hunk 2]
```diff
@@ -40,6 +41,11 @@ def handle(self, domain_name, **options):
...
+            assert hqESQuery().domain(domain_name).run().total == 0, '%s contains data' % hqESQuery.index
```
Comment (orangejenny) [inline corehq/apps/domain/management/commands/delete_domain.py]: Might be nice to extend this to explain what the user needs to do, even just "that should be manually deleted".

Comment (proteusvacuum) [reply corehq/apps/domain/management/commands/delete_domain.py]: Why not just delete it automatically in this script?

**Review by Rohit25negi — APPROVED** [review]

**Comment by nickpell** [comment]
Closing in favor of https://github.com/dimagi/commcare-hq/pull/21964

Since pillows will lag when a lot of data is deleted, this check wouldn't always be reliable anyway.

**Pull request closed without merging**